app/vmselect/prometheus/prometheus.go

```diff
@@ -757,6 +758,111 @@ func LabelsHandler(qt *querytracer.Tracer, startTime time.Time, at *auth.Token,
 	return nil
 }
 
+type metadataResult map[string][]metadataItem
+type metadataItem struct {
+	Type string `json:"type"`
+	Help string `json:"help"`
+	Unit string `json:"unit"`
+}
+
+func MetadataHandler(qt *querytracer.Tracer, startTime time.Time, at *auth.Token, w http.ResponseWriter, r *http.Request) error {
+	cp, err := getCommonParamsForLabelsAPI(r, startTime, false)
+	if err != nil {
+		return err
+	}
+	limit, err := httputil.GetInt(r, "limit")
+	if err != nil {
+		return err
+	}
+	if limit < 0 {
+		limit = 0
+	}
+	limitPerMetric, err := httputil.GetInt(r, "limit_per_metric")
```

Comment:
I don’t think we need to support `limit_per_metric` in addition to `limit`, since most metrics won’t exceed 1 in metadata, and it’s unlikely to exceed 3 or 4.
Prometheus also doesn't support it, https://prometheus.io/docs/prometheus/latest/querying/api/#querying-metadata.

Reply:
And can we keep compatible with Prometheus /api/v1/series API by supporting `match[]=<series_selector>` parameter instead of `metric`?